python/lsst/the/monster/assemble.py

```diff
@@ -0,0 +1,167 @@
+import esutil
+import os
+import numpy as np
+import lsst.utils
+
+from .splinecolorterms import ColortermSpline
+from .refcats import GaiaXPInfo, GaiaDR3Info, SkyMapperInfo, PS1Info, VSTInfo, DESInfo, SynthLSSTInfo
+from .utils import read_stars, makeMonsterSchema, makeMonsterCat
+
+__all__ = ["AssembleMonsterRefcat"]
+
+"""
+This Python script starts with the full Gaia DR3 catalog, then assembles a
+monster refcat by doing the following:
+
+For each shard:
+1. Read the full Gaia DR3 catalog for the shard
+2. Initialize 18 columns (ugrizy fluxes and their errors, and a source flag
+    (integer)) for the results
+(3-6): Within a loop over surveys in order from lowest to highest priority:
+3. Read each of the (already transformed to the DES system) refcats for the
+    shard
+4. Transform each refcat to the (synthetic) LSST system
+5. Match each refcat to the Gaia DR3 catalog
+6. Update the fluxes, flux errors, and flags whenever a value is non-NaN
+"""
+
+
+class AssembleMonsterRefcat:
+    def __init__(self,
+                 gaia_reference_class=GaiaDR3Info,
+                 catalog_info_class_list=[VSTInfo, SkyMapperInfo,
+                                          PS1Info, GaiaXPInfo, DESInfo],
+                 write_path_inp=None,
+                 testing_mode=False,
+                 ):
+
+        self.gaia_reference_info = gaia_reference_class()
+        self.catalog_info_class_list = [cat_info() for cat_info
+                                        in catalog_info_class_list]
+        self.synth_lsst_info = SynthLSSTInfo()
+        self.testing_mode = testing_mode
+        self.write_path_inp = write_path_inp
+        self.all_bands = ['u', 'g', 'r', 'i', 'z', 'y']
+
+    def run(self,
+            *,
+            htmid,
+            verbose=False,
+            ):
+        """Match catalogs to Gaia and transform them to 'the_monster'
+           reference frame.
+
+        Parameters
+        ----------
+        htmid : `int`
+            HTM id of the catalogs.
+        """
+
+        # Read in the Gaia stars in the htmid.
+        gaia_stars_all = read_stars(self.gaia_reference_info.path, [htmid],
+                                    allow_missing=self.testing_mode)
+
+        # Initialize output columns for the fluxes and flux errors,
+        # with all of them set to NaN by default. Also initialize flag columns
+        # with "-1" values as default.
+        nan_column = np.full(len(gaia_stars_all["id"]), np.nan)
+        int_column = np.full(len(gaia_stars_all["id"]), -1)
+
+        for band in self.all_bands:
+            gaia_stars_all.add_column(nan_column,
+                                      name=f"monster_lsst_{band}_flux")
```

Comment:
I think we want the `lsst` part of this to be configurable in some way.  I'm not sure where this should live off the top of my head, but we will want to redo this for `latiss` so it would be nice if we start with obvious things being configurable.  (I'm sure we will find some things that were accidentally hard-coded when we do `latiss` but it's a start.)

Reply:
Good point. I think I've figured a way around this, except for the part in line 41 where it initializes `self.synth_lsst_info = SynthLSSTInfo()` (then later where it uses this). Not sure how to leave that flexible, because it would require importing a different Info() class from refcats.py.